### PR TITLE
Clean up text editing code

### PR DESCRIPTION
**Pull request by Xaeroxe**
A few things were bothering me here so I fixed them.

This should be functionally identical to the prior code, just a bit easier on the eyes.  I also cut out an unnecessary Vec allocation 😉

### CODE REVIEW

File: amethyst_ui/src/text.rs [hunk 1]
```diff
@@ -374,98 +340,40 @@ impl<'a> System<'a> for UiKeyboardSystem {
...
                                 .join()
                                 .filter(|&(_, t)| {
                                     t.pixel_x - t.width / 2.0 <= self.mouse_position.0
                                         && t.pixel_x + t.width / 2.0 >= self.mouse_position.0
                                         && t.pixel_y - t.height / 2.0 <= self.mouse_position.1
                                         && t.pixel_y + t.height / 2.0 >= self.mouse_position.1
-                                }).collect::<Vec<_>>();
-                            // In instances of ambiguity we want to select the element with the
-                            // highest Z order, so we need to find the highest Z order value among
-                            // eligible elements.
```
Comment (jojolepro) [inline amethyst_ui/src/text.rs]: Can we keep the doc?

Comment (Xaeroxe) [reply amethyst_ui/src/text.rs]: Sure

Comment (jojolepro) [reply amethyst_ui/src/text.rs]: thanks :)

**Pull request merged**